preprocessing/nextclade/src/loculus_preprocessing/embl.py

```diff
@@ -0,0 +1,263 @@
+import logging
+from enum import Enum
+from typing import Any
+
+from Bio.Seq import Seq
+from Bio.SeqFeature import CompoundLocation, FeatureLocation, Reference, SeqFeature
+from Bio.SeqRecord import SeqRecord
+
+from loculus_preprocessing.datatypes import ProcessedMetadata, SegmentName
+
+from .config import Config
+
+logger = logging.getLogger(__name__)
+
+# EMBL allowed qualifiers constant
+EMBL_ANNOTATIONS: dict[str, list[str]] = {
+    "cds_qualifiers": [
+        "allele",
+        "artificial_location",
+        "circular_RNA",
+        "codon_start",
+        # "db_xref",  # protein accession of reference
+        "EC_number",
+        "exception",
+        "experiment",
+        "function",
+        "gene",
+        "gene_synonym",
+        "inference",
+        # "locus_tag",  # must be pre-registered with ENA
+        # "old_locus_tag",
+        "map",
+        "note",
+        "number",
+        "operon",
+        "product",
+        "protein_id",
+        "pseudo",
+        "pseudogene",
+        "ribosomal_slippage",
+        "standard_name",
+        "translation",
+    ],
+    "gene_qualifiers": [
+        "allele",
+        # "db_xref",
+        "experiment",
+        "function",
+        "gene",
+        "gene_synonym",
+        "inference",
+        # "locus_tag",
+        # "old_locus_tag",
+        "map",
+        "note",
+        "operon",
+        "product",
+        "pseudo",
+        "pseudogene",
+        "pseudotype",
+        "standard_name",
+        "trans_splicing",
+    ],
+}
+
+
+def get_country(metadata: ProcessedMetadata, config: Config) -> str:
+    country: str = str(metadata.get(config.embl.country_property, "Unknown"))
+    admin_levels = config.embl.admin_level_properties
+    admin: str = ", ".join([metadata.get(level) for level in admin_levels if metadata.get(level)])  # type: ignore
+    return f"{country}: {admin}" if admin else country
+
+
+def get_description(accession, version, db_name) -> str:
+    return (
+        f"Original sequence submitted to {db_name} with accession: {accession}, version: {version}"
```

Comment:
I just realized - this description will have to be modified for all sequences that we ingest from the INSDC and we should probably also modify it if there is a gisaid accession listed